Warnings during build-from-VisualStudio

We get lots of warnings and errors from a build of VisualFSharp.sln from Visual Studio.

@brettfo Would be great to get this clean.  Some are due to multi-proc build.


Errors:
```
1>FSharp.UIResources -> C:\GitHub\dsyme\visualfsharp\vsintegration\src\FSharp.UIResources\bin\Debug\FSharp.UIResources.dll
1>The system cannot find the path specified.
1>C:\GitHub\dsyme\visualfsharp\build\targets\ConvertPortablePdbs.targets(17,5): error MSB3073: The command ""C:\Users\dsyme\.nuget\packages\Microsoft.DiaSymReader.Pdb2Pdb\1.1.0-roslyn-62714-01\tools\Pdb2Pdb.exe" "C:\GitHub\dsyme\visualfsharp\vsintegration\src\FSharp.UIResources\bin\Debug\FSharp.UIResources.dll" /out "C:\GitHub\dsyme\visualfsharp\Debug\net40\bin\ConvertedPdbs\FSharp.UIResources.pdb" /verbose /srcsvrvar SRC_INDEX=public" exited with code 3.
1>Done building project "FSharp.UIResources.csproj" -- FAILED.
```
and
```
2>C:\GitHub\dsyme\visualfsharp\FSharp.Directory.Build.targets(32,5): error MSB3027: Could not copy "C:\GitHub\dsyme\visualfsharp\src\..\Debug\net40\bin\UnitTests\MockTypeProviders\FSharp.Core.dll" to "C:\GitHub\dsyme\visualfsharp\Debug\net40\bin\FSharp.Core.dll". Exceeded retry count of 10. Failed. 
2>C:\GitHub\dsyme\visualfsharp\FSharp.Directory.Build.targets(32,5): error MSB3021: Unable to copy file "C:\GitHub\dsyme\visualfsharp\src\..\Debug\net40\bin\UnitTests\MockTypeProviders\FSharp.Core.dll" to "C:\GitHub\dsyme\visualfsharp\Debug\net40\bin\FSharp.Core.dll". The requested operation cannot be performed on a file with a user-mapped section open.
2>C:\GitHub\dsyme\visualfsharp\FSharp.Directory.Build.targets(32,5): error MSB3021: 
```
and
```
21>------ Build started: Project: FSharp.Compiler.UnitTests, Configuration: Debug Any CPU ------
21>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(2052,5): warning MSB3277: Found conflicts between different versions of "System.ValueTuple" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
```
and
```
15>------ Build started: Project: FSharp.VS.FSI, Configuration: Debug Any CPU ------
15>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(4289,5): warning MSB3026: Could not copy "C:\GitHub\dsyme\visualfsharp\Debug\net40\bin\FSharp.Compiler.Private.pdb" to "bin\Debug\FSharp.Compiler.Private.pdb". Beginning retry 1 in 1000ms. The process cannot access the file 'C:\GitHub\dsyme\visualfsharp\Debug\net40\bin\FSharp.Compiler.Private.pdb' because it is being used by another process.
```
and
```
15>FSharp.VS.FSI -> C:\GitHub\dsyme\visualfsharp\vsintegration\src\FSharp.VS.FSI\bin\Debug\FSharp.VS.FSI.dll
15>C:\Users\dsyme\.nuget\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error : CreatePkgDef : error : FileLoadException: Could not load file or assembly 'Microsoft.VisualStudio.Shell.Framework, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The process cannot access the file because it is being used by another process. (Exception from HRESULT: 0x80070020)
15>C:\Users\dsyme\.nuget\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error :    at Microsoft.VisualStudio.Tools.CreatePkgDef.DoCreatePkgDef(InputArguments inputArguments)
15>C:\Users\dsyme\.nuget\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error :    at Microsoft.VisualStudio.Tools.CreatePkgDef.Main(String[] arguments)
15>C:\Users\dsyme\.nuget\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error : FileLoadException: Could not load file or assembly 'FSharp.Compiler.Private.dll' or one of its dependencies. The process cannot access the file because it is being used by another process. (Exception from HRESULT: 0x80070020)
15>C:\Users\dsyme\.nuget\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error :    at System.Reflection.AssemblyName.nGetFileInformation(String s)
15>C:\Users\dsyme\.nuget\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error :    at System.Reflection.AssemblyName.GetAssemblyName(String assemblyFile)
15>C:\Users\dsyme\.nuget\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error :    at Microsoft.VisualStudio.Tools.CreatePkgDef.<AssemblyResolveHandler>g__LoadAssemblyIfItMatchesArgs7_0(String pathToAssembly, <>c__DisplayClass7_0& )
15>C:\Users\dsyme\.nuget\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error :    at Microsoft.VisualStudio.Tools.CreatePkgDef.AssemblyResolveHandler(Object sender, ResolveEventArgs args)
15>C:\Users\dsyme\.nuget\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error :    at System.AppDomain.OnAssemblyResolveEvent(RuntimeAssembly assembly, String assemblyFullName)
15>C:\Users\dsyme\.nuget\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error : FileLoadException: The process cannot access the file because it is being used by another process. (Exception from HRESULT: 0x80070020)
15>C:\Users\dsyme\.nuget\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error : 
15>C:\Users\dsyme\.nuget\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error : 
15>C:\Users\dsyme\.nuget\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error :
15>Done building project "FSharp.VS.FSI.fsproj" -- FAILED.
```
and
```
30>C:\Users\dsyme\.nuget\packages\xlifftasks\0.2.0-beta-000081\build\XliffTasks.targets(122,5): error : Two or more source files to be translated in the same project are named FSData.resx. Give them unique names or set unique XlfTranslatedFilename metadata.
```

Warnings:

```
1>AdvancedOptionsControl.xaml.cs(23,16,23,38): warning CS1591: Missing XML comment for publicly visible type or member 'AdvancedOptionsControl.AdvancedOptionsControl()'
1>IntegerRangeValidationRule.cs(8,18,8,44): warning CS1591: Missing XML comment for publicly visible type or member 'IntegerRangeValidationRule'
1>IntegerRangeValidationRule.cs(10,16,10,42): warning CS1591: Missing XML comment for publicly visible type or member 'IntegerRangeValidationRule.IntegerRangeValidationRule()'
1>IntegerRangeValidationRule.cs(15,20,15,23): warning CS1591: Missing XML comment for publicly visible type or member 'IntegerRangeValidationRule.Min'
1>IntegerRangeValidationRule.cs(17,20,17,23): warning CS1591: Missing XML comment for publicly visible type or member 'IntegerRangeValidationRule.Max'
1>IntegerRangeValidationRule.cs(19,42,19,50): warning CS1591: Missing XML comment for publicly visible type or member 'IntegerRangeValidationRule.Validate(object, CultureInfo)'
```
and
```
26>------ Build started: Project: FSharp.LanguageService, Configuration: Debug Any CPU ------
26>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(2052,5): warning MSB3277: Found conflicts between different versions of "Microsoft.Build" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
26>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(2052,5): warning MSB3277: Found conflicts between different versions of "Microsoft.DiaSymReader" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
26>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(2052,5): warning MSB3277: Found conflicts between different versions of "Microsoft.Build.Tasks.Core" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
26>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(2052,5): warning MSB3277: Found conflicts between different versions of "Microsoft.Build.Framework" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
26>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(2052,5): warning MSB3277: Found conflicts between different versions of "Microsoft.VisualStudio.Threading" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
```

OK, the current problems I see are:

1. You need to `build vs debug test` before or else you get errors like this:

       43>------ Build started: Project: VisualFSharp.UnitTests, Configuration: Debug Any CPU ------
       43>C:\Program Files\dotnet\sdk\2.1.100\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDependencyResolution.targets(327,5): error : Assets file 'C:\GitHub\dsyme\visualfsharp\vsintegration\tests\unittests\obj\project.assets.json' not found. Run a NuGet package restore to generate this file.
       43>C:\Program Files\dotnet\sdk\2.1.100\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDependencyResolution.targets(167,5): error : Assets file 'C:\GitHub\dsyme\visualfsharp\vsintegration\tests\unittests\obj\project.assets.json' not found. Run a NuGet package restore to generate this file.

   I suppose we should restore those test projects.

2. Rebuilding always rebuilds `FSharp.Editor.dll` because of this:

```
Target "GenerateAssemblyInfo" in file "C:\Program Files\dotnet\sdk\2.1.100\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.GenerateAssemblyInfo.targets" from project "C:\GitHub\dsyme\visualfsharp\vsintegration\src\FSharp.Editor\FSharp.Editor.fsproj" (target "CoreCompile" depends on it):
Done building target "GenerateAssemblyInfo" in project "FSharp.Editor.fsproj".
Target "CoreCompile" in file "C:\GitHub\dsyme\visualfsharp\Proto\net40\bin\Microsoft.FSharp.Targets" from project "C:\GitHub\dsyme\visualfsharp\vsintegration\src\FSharp.Editor\FSharp.Editor.fsproj" (target "Compile" depends on it):
Building target "CoreCompile" completely.
Input file "obj\debug\FSharp.Editor.resources" is newer than output file "obj\debug\FSharp.Editor.xml".
```

   It looks to me like the ``HACK_CopyOutputsToTheProperLocation`` is always copying that file across?  But why is it copying from `debug\net40\obj\` to `debug\net40\obj\`?


```
  Copying file from "obj\debug\FSharp.Editor.resources" to "C:\GitHub\dsyme\visualfsharp\debug\net40\obj\FSharp.Editor.resources".
```


3. I get this

```
14>ProjectSystem.Base -> C:\GitHub\dsyme\visualfsharp\vsintegration\src\FSharp.ProjectSystem.Base\Project\bin\Debug\FSharp.ProjectSystem.Base.dll
14>C:\GitHub\dsyme\visualfsharp\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error : CreatePkgDef : error : ProvideEditorFactoryAttribute: Could not load file or assembly 'Microsoft.VisualStudio.Shell.15.0, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The process cannot access the file because it is being used by another process. (Exception from HRESULT: 0x80070020)
14>C:\GitHub\dsyme\visualfsharp\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error :
```

4. I get this:

```
14>ProjectSystem.Base -> C:\GitHub\dsyme\visualfsharp\vsintegration\src\FSharp.ProjectSystem.Base\Project\bin\Debug\FSharp.ProjectSystem.Base.dll
15>LanguageServiceProfiling -> C:\GitHub\dsyme\visualfsharp\vsintegration\Utils\LanguageServiceProfiling\bin\Debug\LanguageServiceProfiling.exe
14>C:\GitHub\dsyme\visualfsharp\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error : CreatePkgDef : error : FileLoadException: Could not load file or assembly 'Microsoft.VisualStudio.Shell.Framework, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The process cannot access the file because it is being used by another process. (Exception from HRESULT: 0x80070020)
14>C:\GitHub\dsyme\visualfsharp\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error :    at Microsoft.VisualStudio.Tools.CreatePkgDef.DoCreatePkgDef(InputArguments inputArguments)
14>C:\GitHub\dsyme\visualfsharp\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error :    at Microsoft.VisualStudio.Tools.CreatePkgDef.Main(String[] arguments)
14>C:\GitHub\dsyme\visualfsharp\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error : FileLoadException: Could not load file or assembly 'FSharp.Compiler.Private.dll' or one of its dependencies. The process cannot access the file because it is being used by another process. (Exception from HRESULT: 0x80070020)
14>C:\GitHub\dsyme\visualfsharp\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error :    at System.Reflection.AssemblyName.nGetFileInformation(String s)
14>C:\GitHub\dsyme\visualfsharp\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error :    at System.Reflection.AssemblyName.GetAssemblyName(String assemblyFile)
14>C:\GitHub\dsyme\visualfsharp\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error :    at Microsoft.VisualStudio.Tools.CreatePkgDef.<AssemblyResolveHandler>g__LoadAssemblyIfItMatchesArgs7_0(String pathToAssembly, <>c__DisplayClass7_0& )
14>C:\GitHub\dsyme\visualfsharp\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error :    at Microsoft.VisualStudio.Tools.CreatePkgDef.AssemblyResolveHandler(Object sender, ResolveEventArgs args)
14>C:\GitHub\dsyme\visualfsharp\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error :    at System.AppDomain.OnAssemblyResolveEvent(RuntimeAssembly assembly, String assemblyFullName)
14>C:\GitHub\dsyme\visualfsharp\packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targets(979,5): error : FileLoadException: The process cannot access the file because it is being used by another process. (Exception from HRESULT: 0x80070020)
```

5.  I get this

```
22>------ Build started: Project: VisualFSharp.UnitTests, Configuration: Debug Any CPU ------
22>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(2052,5): warning MSB3277: Found conflicts between different versions of "Microsoft.DiaSymReader" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
22>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(2052,5): warning MSB3277: Found conflicts between different versions of "Microsoft.VisualStudio.Threading" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
22>C:\GitHub\dsyme\visualfsharp\packages\xlifftasks\0.2.0-beta-000081\build\XliffTasks.targets(122,5): error : Two or more source files to be translated in the same project are named FSData.resx. Give them unique names or set unique XlfTranslatedFilename metadata.
22>Done building project "VisualFSharp.UnitTests.fsproj" -- FAILED.
```

--- a/vsintegration/src/FSharp.ProjectSystem.FSharp/ProjectSystem.fsproj
+++ b/vsintegration/src/FSharp.ProjectSystem.FSharp/ProjectSystem.fsproj
@@ -23,6 +23,7 @@
   <ItemGroup>
     <VSCTCompile Include="MenusAndCommands.vsct">
       <XlfInput>false</XlfInput>
+      <ResourceName>Menus.ctmenu</ResourceName>
     </VSCTCompile>
     <EmbeddedResource Update="VSPackage.resx">
       <GenerateSource>true</GenerateSource>

--- a/vsintegration/src/FSharp.UIResources/FSharp.UIResources.csproj
+++ b/vsintegration/src/FSharp.UIResources/FSharp.UIResources.csproj
@@ -7,6 +7,7 @@
     <OutputType>Library</OutputType>
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <RootNamespace>Microsoft.VisualStudio.FSharp.UIResources</RootNamespace>
+    <NoWarn>$(NoWarn);1591</NoWarn>
   </PropertyGroup>
 
   <ItemGroup>

